### DISCUSSION TITLE
A question about 'number of steps' in input.json

### DISCUSSION THREAD

**Discussion by kaijiz** [deepmodeling/deepmd-kit#3660] [Q&A] [answered]
Hi, i am a little confused how to set the number of 'number of steps', is it same as 'number of epoch'? For example, if i have 10000 training data, and i use batch_size=1 which should mean each training data is a batch. So now i have 100000 batches, can i choose 'number of step' to be larger than 10000. If i don't consider the calculation cost, is this value the bigger, the better? Thanks for help.

**Comment by njzjz** [comment, marked as answer]
n_step * batch_size = n_epoch * n_data

> If i don't consider the calculation cost, is this value the bigger, the better?

Yes, more is better.
